openlabcmd/openlabcmd/zk.py

```diff
@@ -84,19 +84,28 @@ def lost(self):
             return True
         return self.client.state == KazooState.LOST
 
-    def connect(self, hosts=None, read_only=False):
-        if not hosts:
+    def connect(self, hosts=None, timeout=None, read_only=False):
+        if not hosts or not timeout:
             if not self.config:
                 raise exceptions.ClientError('Either config object or hosts '
                                              'string should be provided.')
-            else:
-                try:
-                    hosts = self.config.get('ha', 'zookeeper_hosts')
-                except configparser.NoOptionError:
-                    raise exceptions.ClientError(
-                        "The config doesn't [ha]zookeeper_hosts option.")
+            try:
+                hosts = hosts or self.config.get('ha', 'zookeeper_hosts')
+            except configparser.NoOptionError:
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_hosts option.")
+            try:
+                timeout = timeout or self.config.get('ha', 'zookeeper_timeout')
+                timeout = int(timeout)
+            except configparser.NoOptionError:
+                raise exceptions.ClientError(
+                    "The config doesn't contain [ha]zookeeper_timeout option.")
```

Comment:
why this is mandatory ？ does it has default value ?

Reply:
if so, does the `config` object can define config options with type limit, required, and default value ?

Reply:
1. Done, default is 5.
2. I didn't find a way to do it.